SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialTraceEventBuilder.h

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Interop/Connection/SpatialTraceEvent.h"
+#include "SpatialCommonTypes.h"
+#include "WorkerSDK/improbable/c_worker.h"
+
+#define GDK_EVENT_NAMESPACE "unreal_gdk."
+
+namespace SpatialGDK
+{
+class FSpatialTraceEventBuilder
+{
+public:
+	FSpatialTraceEventBuilder(const char* InType);
+	FSpatialTraceEventBuilder(const char* InType, const FString& InMessage);
+
+	// --- Builder Functions ---
+
+	FSpatialTraceEventBuilder AddObject(const UObject* Object);
+	FSpatialTraceEventBuilder AddObject(const FString& Key, const UObject* Object);
+	FSpatialTraceEventBuilder AddFunction(const UFunction* Function);
+	FSpatialTraceEventBuilder AddEntityId(const Worker_EntityId EntityId);
+	FSpatialTraceEventBuilder AddComponentId(const Worker_ComponentId ComponentId);
+	FSpatialTraceEventBuilder AddFieldId(const uint32 FieldId);
+	FSpatialTraceEventBuilder AddNewWorkerId(const uint32 NewWorkerId);
+	FSpatialTraceEventBuilder AddCommand(const FString& Command);
+	FSpatialTraceEventBuilder AddRequestID(const int64 RequestID);
+	FSpatialTraceEventBuilder AddNetRole(const ENetRole Role);
+	FSpatialTraceEventBuilder AddKeyValue(FString Key, FString Value);
+	FSpatialTraceEvent Get() &&;
```

Comment:
Nit - would possibly make this name a bit longer as Get is often used to get a non owning pointer to underlying data. I think when used in the builder context it's normally fairly obvious what it means. But something like `CreateEvent` might make it a bit clearer.